jaxrs-api/src/main/java/jakarta/ws/rs/core/Cookie.java

```diff
@@ -216,4 +234,123 @@ public boolean equals(final Object obj) {
         }
         return true;
     }
+
+    /**
+     * JAX-RS {@link Cookie} builder class.
+     * <p>
+     * Cookie builder provides methods that let you conveniently configure and subsequently build a new
+     * {@code Cookie} instance.
+     * </p>
+     * For example:
+     *
+     * <pre>
+     * Cookie cookie = new Cookie.Builder("name")
+     *         .path("/")
+     *         .domain("domain.com")
+     *         .build();
+     * </pre>
+     * <p>
+     * It should be preferred over any {@code Cookie} telescoping constructors that have following disadvantages:
```

Comment:
Well I fully agree with you. 
But you asked me to explain why the builder is good for here and you wanted me to let user discover it in the doc..so that's what I did. 
Sorry if I do not undertsand what you meant.

Reply:
That's all fine. It's just the word "SHOULD" that bothers me in a specification project, as there are rules when to use SHOULD, MUST, etc.